components/raftstore-v2/src/operation/ready/mod.rs

```diff
@@ -298,6 +299,11 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             }
         }
 
+        // check whether pending peers has caught up.
+        if self.any_new_peer_catch_up(from_peer_id) {
```

Comment:
There are two different cases. If the peer is pending, then only `AppendResponse` can bring it back to not pending. If the peer is down, then `HeartbeatResponse` and `AppendResponse` can bring it back to up.

Reply:
What I mean is only need to call it when the message type is `AppendResponse`.